### ISSUE TITLE
An unexpected behavior of  PostgaphiQL, maybe bug

### ISSUE THREAD

**Issue by geraldurbas** [graphile/crystal#2283]
Dear Graphile Team,

I stumbled upon the following behavior in an inherited project. That doesn't seem directly logical to me. In any case, the whole thing is already very cool and, as promised, super fast.
thanks!

3 Queries - quite self explaining.
That's not how it works as expected:

```
query MyQueryfails {
  files(
    filter: {or: {mimeType: {like: "%png%"}, or: {mimeType: {like: "%jpeg%"}}}}
  ) {
    nodes {
      codec
      bitrate
      contentSize
      mimeType
    }
  }
}
```

Here's how it works:

```
query MyQueryworks {
  files(
    filter: {or: [{mimeType: {like: "%png%"}},{mimeType: {like: "%jpeg%"}}]}
  ) {
    nodes {
      codec
      bitrate
      contentSize
      mimeType
    }
  }
}
```

so far so good

But why is it working again?

```
query WhyIstheQueryWorking {
  files(
    filter: {or: {contentSize: {greaterThan: 70000}, or: {contentSize: {lessThan: 80000}}}}
  ) {
    nodes {
      codec
      bitrate
      contentSize
	  mimeType
    }
  }
}
```

Finally, this beautiful automatically generated GUI also seems to have trouble with the array in the or option.

![image](https://github.com/user-attachments/assets/074f2eb3-0b6c-47cb-9e0a-52d5e2cb76d3)

**Comment by benjie** [comment]
Yeah this is an issue with graphiql explorer that is yet to be addressed:

https://github.com/OneGraph/graphiql-explorer/issues/32

As for why it works without a list:

https://spec.graphql.org/draft/#sec-List.Input-Coercion

**Comment by geraldurbas** [comment]
> Yeah this is an issue with graphiql explorer that is yet to be addressed:
> 
> [OneGraph/graphiql-explorer#32](https://github.com/OneGraph/graphiql-explorer/issues/32)
> 
> As for why it works without a list:
> 
> https://spec.graphql.org/draft/#sec-List.Input-Coercion

Thanks for the reference link and fast response.
